cpp/src/join/join.cu

```diff
@@ -314,11 +314,13 @@ std::unique_ptr<experimental::table> construct_join_output_df(
       auto common_from_right = experimental::detail::gather(right.select(right_common_col),
                                                             complement_indices.second.begin(),
                                                             complement_indices.second.end(),
-                                                            nullify_out_of_bounds);
+                                                            nullify_out_of_bounds,
+                                                            mr);
```

Comment:
As @davidwendt mentioned, think you have to pass stream as well.

Reply:
Just pushed the change to pass stream as well.